src/CentrifugeRouter.sol

```diff
@@ -140,6 +150,18 @@ contract CentrifugeRouter is Auth, ICentrifugeRouter {
         IERC7540Vault(vault).deposit(maxDeposit, receiver, controller);
     }
 
+    /// @inheritdoc ICentrifugeRouter
+    function cancelDepositRequest(address vault, address controller) external payable protected {
+        validateController(vault, controller);
```

Comment:
Do we need `validateController` check in the route at all ? It's a duplicate and it's already being done at ERC7540 level :S

Reply:
`validateController` gets called at the vault level, but then it's checking whether `router` is a valid controller. It's endorsed so it always passes this check, so without also validating at the `router` level anybody could call this on anyone else's behalf

Reply:
This doesn't work. For `cancelDeposit/RedeemRequest`, it doesn't make sense to pass a `controller`, it should just always use `_initiatior()`

Reply:
For `claimCancelDeposit/RedeemRequest`, you should copy the logic from `claimDeposit/RedeemRequest`:
```solidity
require(
            controller == _initiator || (controller == receiver && opened[controller][vault] == true),
            "CentrifugeRouter/invalid-sender"
        );
```

Reply:
> For cancelDeposit/RedeemRequest, it doesn't make sense to pass a controller, it should just always use _initiatior()

@AStox see this line above, that is not addressed correctly now

Reply:
Yeah exactly